frontend/src/lib/RunUtils.ts

```diff
@@ -63,6 +63,25 @@ function getFirstExperimentReferenceId(run?: ApiRun | ApiJob): string | null {
   return null;
 }
 
+export interface ExperimentReferenceInfo {
+  name: string;
+  id: string;
+}
+
+function getFirstExperimentReferenceInfo(run?: ApiRun | ApiJob): ExperimentReferenceInfo | null {
+  if (run) {
+    const reference = getAllExperimentReferences(run)[0];
+    if (reference && reference.name && reference.key && reference.key.id) {
```

Comment:
I am assuming backend should always provide the two together. Do you think it's a fair assumption?

Reply:
There is a possibility that the name is not provided when an older reference or lower version backend is used. Additionally, it is possible that the first experiment reference returned by `getAllExperimentReferences` is not a valid `ExperimentReferenceInfo` object. Due to this, all indexes of the array returned by `getAllExperimentReferences` should be checked to ensure that at least one is a valid `ExperimentReferenceInfo` object. This was discussed offline with @rileyjbauer and @IronPan.